docs/proposals/enable-local-source-code-support.md

```diff
@@ -0,0 +1,301 @@
+
+The Build resource API needs to provide means of signalizing the desire of local source code. In terms of API changes this should be minimal.
+
+Therefore, users should be able to signalize this feature via:
+
+```yaml
+spec:
+  source:
+    container:
+      image: docker.io/<registry-repository>/abundle@sha256:3235326357dfb65f17 # Note: full digest incomplete
+```
+
+#### As a Shipwright User I want to build images in Development mode from my local Dockerfile or plain source code
+
+Users should only need to specify the mandatory fields under `spec.source.container` without modifying their existing strategies of choice. Local source code feature should work out of the box with different tools, like Kaniko, BuildKit, Buildpacks, et al.
+
+This will be done via Shipwright/CLI in the form of:
+
+```sh
+shp build create a-build --local-path "/absolute-path/src/github.com/shipwright-io/sample-go/docker-build" --output-image "docker.io/<registry-repository>/img:latest" --strategy "kaniko"
+```
+
+_Note_: The above is just an example on how the CLI might look like.
+
+#### As a Shipwright User I want to ensure reproducible bundle images
+
+We want to provide trust to users when building bundle images with their source code, so that on multiple builds with the same local source code, the bundle image always express the same _SHA_. Subsequent Builds should generate the same image if the source code is unchanged.
+
+#### As a Shipwright User I want to avoid excessive registry storage from Bundle images
+
+We want to allow users to signalize the desire of pruning a bundle image that was used for building their application container image.
+
+Adding `pruneBundle` should ensure a mechanism for deleting `docker.io/<registry-repository>/bundle:latest`.
+
+```yaml
+---
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: a-kaniko-build
+spec:
+  source:
+    container:
+      image: docker.io/<registry-repository>/bundle:latest
+      pruneBundle: true # Disable by default
+  strategy:
+    name: kaniko
+    kind: ClusterBuildStrategy
+  output: ...
+```
+
+The above Build definition should ensure that a step in our [TaskSpec](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/reconciler/buildrun/resources/taskrun.go#L59) definition is appended, which will ensure a pruning of the image in the related container registry.
+
+_Note_: Supporting multiple registry providers might be too complicated and different alternatives should be considering for pruning.
+
+#### As a Shipwright/CLI contributor I want to provide a mechanism to support local source code
+
+This is related to the CLI workflow, that should support the bundle approach and maintain the bundle base image. The implementation should be generic enough for other workflows, like OpenShift Build, IBM Cloud Code Engine, to consume it.
+
+### Implementation Details/Notes/Constraints [optional]
```

Comment:
I'd like to see a description of the end-to-end workflow here, which shows:

1. How a developer declares that their `Build` object uses local source.
2. How a developer starts a build that uses local source from the command line.
3. What the cli does to convert the `Build` object to a `BuildRun` that references the pushed image digest.
4. How the build cleans up the build if `pruneBundle` is set to true.

I think the pieces of this are elsewhere in the EP and referenced repos. Having it all together in one place helps tie the narrative together. Likewise, I'd like to see the updated APIs in full, and in context with the relevant Shipwright objects.

Reply:
I will add that soon.

Reply:
done